src/statsmodels/formula.jl

```diff
@@ -285,11 +285,7 @@ end
 
 const DEFAULT_CONTRASTS = DummyCoding
 
-function ModelFrame(trms::Terms, d::AbstractDataFrame;
-                    contrasts::Dict = Dict())
-    df, msng = na_omit(DataFrame(map(x -> d[x], trms.eterms)))
-    names!(df, convert(Vector{Symbol}, map(string, trms.eterms)))
-    for c in eachcol(df) dropunusedlevels!(c[2]) end
+function evalcontrasts(df::AbstractDataFrame, contrasts::Dict = Dict())
 
```

Comment:
And BTW, move the comments below before the signature, as they clearly document the function.